### DISCUSSION TITLE
Aux Port as Power Input

### DISCUSSION THREAD

**Discussion by hectorvanzyl** [slipx06/sunsynk-power-flow-card#315] [Sunsynk/Deye]
Sunsynk/Deye inverters have option to use Aux port for Micro Inverter/Generator power input, using the card in such a setup obviously cause energy value issues, I assume support for Energy input on Aux port is not supported yet?
![PV Input on Aux Port](https://github.com/slipx06/sunsynk-power-flow-card/assets/51006824/4124700a-e67d-448a-a5a4-ff9dbcb691af)

**Comment by slipx06** [comment]
Hi. What issues are you experiencing?

**Comment by hectorvanzyl** [comment]
Hi, I would not say an issue, maybe feature request. Aux Options are currently under Load Options, but on Sunsynk/Deye Aux can also function as a Micro Inverter or Generator input receiving energy instead of supplying. In the above image the solar is actually micro inverter energy imput on the Aux port. Home load is 753W with 2kW exported, the 3.77kW calculation is thus wrong. I trust this makes more sense, English is my 2nd language. Thanks for the reply and awesome integration nonetheless.

**Comment by hectorvanzyl** [comment]
PS. I don't think there is many people using the Aux port as an Energy input port since I could not find any such references, I don't know whethher it'll be worth investing your time in a seldom used feature such as my request. The HomeAssistant enegy dashboard do work correctly with it but obviously it's very simplistic. Thanks

**Reply by higgy-de** [reply]
In Deye Forum and german Forum there are many many users who use the MI Port as it, like Deye supportetd ...( me too ) would be nice if you can add an " 5. tracker" for this ...

**Comment by slipx06** [comment]
Hi. Currently the card can be used to display AUX input or output. By default Energy received is shown as a negative number as you have it on the card.  If you do not provide a sensor for essential_power: then the card will calculate it using the formula below

`essential = inverter_power_175 + grid_power_169 - aux_power_166`

It looks like you have not defined a sensor for grid_power_169 or the sensor is wrong as it is showing as 0?
If you have a sensor for your house load you can also use that for your essential_power: in the config instead of using the formula.

**Comment by hectorvanzyl** [comment]
Thanks a lot, I will verify all sensors and check again, best wishes.
